src/k8s_kube_system.py

```diff
@@ -51,17 +51,22 @@ def get_kube_system_pods_not_running(charm) -> Optional[List]:
     # should not be considered as pending Pods.
     valid_phases = ["Running", "Succeeded", "Failed"]
 
+    # Pods in ['Succeeded', 'Failed'] phases are not meant to become ready
+    def is_yet_to_be_ready(pod):
+
+        if pod["status"]["phase"] == "Running":
```

Comment:
We can use an early return, just to avoid the unnecessary indentation: 
```suggestion
    def is_yet_to_be_ready(pod):
        if pod["status"]["phase"] != "Running":
            return False
```